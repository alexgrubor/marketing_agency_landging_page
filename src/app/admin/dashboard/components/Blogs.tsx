'use client'
import { useState } from "react"
const Blogs = () => {
  const initialBlogs = [
    {
      id: 1,
      title: "10 Tips for Effective Content Marketing",
      author: "John Doe",
      date: "March 10, 2024",
      content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit...",
    },
    {
      id: 2,
      title: "The Importance of SEO in Digital Marketing",
      author: "Jane Smith",
      date: "March 15, 2024",
      content: "Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua...",
    },
    {
      id: 3,
      title: "Social Media Marketing Strategies for 2024",
      author: "Alex Johnson",
      date: "March 20, 2024",
      content: "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris...",
    }
  ];

  const [blogs, setBlogs] = useState(initialBlogs);
  const removeBlog = (id: number) => {
    const updatedBlogs = blogs.filter(blog => blog.id !== id);
    setBlogs(updatedBlogs);
  };
  return (
    <div className="container mx-auto px-4 py-8">
      <h1 className="text-2xl font-bold mb-4">Manage Blogs</h1>
      <div>
    
        {blogs.map(blog => (
          <div key={blog.id} className="bg-secondBg shadow-md rounded-md p-6 mb-4">
            <h2 className="text-xl font-semibold mb-2">{blog.title}</h2>
            <p><strong>Author:</strong> {blog.author}</p>
            <p><strong>Date:</strong> {blog.date}</p>
            <p className="mt-2">{blog.content}</p>
       
            <div className="mt-4 flex justify-end">
              <button
                className="mr-2 bg-red-500 hover:bg-red-600 text-white font-bold py-2 px-4 rounded"
                onClick={() => removeBlog(blog.id)}
              >
                Delete
              </button>
              <button className="bg-green-500 hover:bg-green-600 text-white font-bold py-2 px-4 rounded">
                <a href={'#'}>Update</a>
              </button>
            </div>
          </div>
        ))}
      </div>
    </div>
  )
}
export default Blogs