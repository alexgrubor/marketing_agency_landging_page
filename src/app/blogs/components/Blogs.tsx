"use client";
import Image from "next/image";
import { useState, useEffect } from "react";
import axios from "axios";

interface Blog {
  id: number;
  title: string;
  body: string;
  reactions?: number;
  tags?: string[];
  userId?: number;
}

const Blogs = () => {
  const [blogs, setBlogs] = useState<Blog[]>([]);
  const [loading, setLoading] = useState(true);
  const [currentPage, setCurrentPage] = useState(1);
  const blogsPerPage = 10;

  useEffect(() => {
    axios.get("https://dummyjson.com/posts").then((res) => {
      setBlogs(res.data.posts);
      setLoading(false);
    });
  }, []);

  const indexOfLastBlog = currentPage * blogsPerPage;
  const indexOfFirstBlog = indexOfLastBlog - blogsPerPage;
  const currentBlogs = blogs.slice(indexOfFirstBlog, indexOfLastBlog);

  const handlePageChange = (pageNumber: number) => {
    setCurrentPage(pageNumber);
  };

  return (
    <div className="bg-secondBg p-4">
      {loading && <div>loading.....</div>}
      
      <div className="flex flex-wrap justify-center gap-4 sm:gap-2 lg:gap-4">
        {currentBlogs.map((blog) => {
             const truncatedBody = blog.body.length > 140 ? blog.body.slice(0, 140) + '...' : blog.body;
             const truncatedTitile = blog.title.length > 20 ? blog.title.slice(0, 20) + '...' : blog.title;
          return (
            <div
              key={blog.id}
              className="bg-myPurple-600  w-full sm:w-1/2 md:w-1/3 lg:w-1/6 p-4 flex flex-col h-full overflow-hidden rounded-lg"
            >
              <h2 className="text-2xl my-2 font-bold">{truncatedTitile}</h2>
              <Image
                src={
                  blog.id % 2 === 0
                    ? "/images/event1.jpg"
                    : "/images/event2.jpg"
                }
                alt="Blog_item"
                width={600}
                height={400}
              />
              <p>{truncatedBody}</p>
              <button className="bg-secondBg py-2 my-2 px-3">Read More</button>
            </div>
          );
        })}
      </div>
      <div className="flex justify-center my-8 gap-3">
        {Array.from(
          { length: Math.ceil(blogs.length / blogsPerPage) },
          (_, i) => (
            <button
              className={`${
                i + 1 === currentPage
                  ? "bg-myPurple-600 text-white"
                  : "bg-white"
              } px-3 py-1 rounded-lg`}
              key={i + 1}
              onClick={() => handlePageChange(i + 1)}
            >
              {i + 1}
            </button>
          )
        )}
      </div>
    </div>
  );
};
export default Blogs;
