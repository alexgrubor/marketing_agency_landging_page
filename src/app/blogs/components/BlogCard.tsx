import { Blog } from "./Blogs";
import Image from "next/image";

interface BlogCardProps {
    blog: Blog;
}
const BlogCard = ({ blog }: BlogCardProps) => {
    const truncatedBody = blog.body.length > 140 ? blog.body.slice(0, 140) + '...' : blog.body;
    const truncatedTitle = blog.title.length > 20 ? blog.title.slice(0, 20) + '...' : blog.title;
  
    return (
      <div
        key={blog.id}
        className="bg-myPurple-600  w-full sm:w-1/2 md:w-1/3 lg:w-1/6 p-4 flex flex-col h-full overflow-hidden rounded-lg"
      >
        <h2 className="text-2xl my-2 font-bold">{truncatedTitle}</h2>
        <Image
          src={blog.id % 2 === 0 ? "/images/event1.jpg" : "/images/event2.jpg"}
          alt="Blog_item"
          width={600}
          height={400}
        />
        <p>{truncatedBody}</p>
        <button className="bg-secondBg py-2 my-2 px-3">Read More</button>
      </div>
    );
}

export default BlogCard;