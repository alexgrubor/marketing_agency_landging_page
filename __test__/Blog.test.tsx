import React from 'react'
import { render, screen } from '@testing-library/react'
import Blog from '../src/app/blogs/page'
import Blogs from '@/app/blogs/components/Blogs';
import BlogCard from '@/app/blogs/components/BlogCard';

import '@testing-library/jest-dom'; 
test('renders the title', () => {
  render(<Blog />)
    const title = screen.getByText(/Blogs/i)
    expect(title).toBeInTheDocument();
})

//snapshot 
test('expect to render BlogCard', () => {
    const blog = {
        id: 1,
        title: 'title',
        body: 'body'
        
    }
    const { container } = render(<BlogCard blog={blog} />)
    expect(container).toMatchSnapshot()

}) 


