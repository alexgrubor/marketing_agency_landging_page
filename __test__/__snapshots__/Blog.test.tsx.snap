// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render BlogCard 1`] = `
<div>
  <div
    class="bg-myPurple-600  w-full sm:w-1/2 md:w-1/3 lg:w-1/6 p-4 flex flex-col h-full overflow-hidden rounded-lg"
  >
    <h2
      class="text-2xl my-2 font-bold"
    >
      title
    </h2>
    <img
      alt="Blog_item"
      data-nimg="1"
      decoding="async"
      height="400"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fevent2.jpg&w=1200&q=75"
      srcset="/_next/image?url=%2Fimages%2Fevent2.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fevent2.jpg&w=1200&q=75 2x"
      style="color: transparent;"
      width="600"
    />
    <p>
      body
    </p>
    <button
      class="bg-secondBg py-2 my-2 px-3"
    >
      Read More
    </button>
  </div>
</div>
`;
